### PR TITLE
Download all feature

### PR DESCRIPTION
**Pull request by KWMORALE**
**Your checklist for this pull request**
- [x] I've read the [contributing guideline](https://github.com/CERT-Polska/mquery/blob/master/CONTRIBUTING.md).
- [x] I've tested my changes by building and running mquery, and testing changed functionality (if applicable)
- [x] add the download button to matches column

**What is the new behaviour?**
the ability to download all hashes or matches by clicking one button

**Closing issues**
#33

### CODE REVIEW

**Comment by KWMORALE** [comment]
Screenshot from 2020-05-12 17-11-55

Add the moment it looks like this.
In coming days I will working on add the download button to matches and returning ziping matched files.

File: src/app.py [hunk 1]
```diff
@@ -62,6 +62,21 @@ def download(job_id: str, ordinal: str, file_path: str) -> Any:
     return FileResponse(file_path, filename=attach_name + ext + "_")
 
 
+@app.get("/api/download/hashes/{hash}")
+def download_hashes(
+    hash: str, offset: int = 0, limit: int = Query(...)
+) -> Any:
+    hashes = ",\n".join(
```
Comment (msm-code) [inline src/app.py:69]: definitely no `,`, just newline is enough

File: src/app.py [hunk 2]
```diff
@@ -62,6 +62,21 @@ def download(job_id: str, ordinal: str, file_path: str) -> Any:
     return FileResponse(file_path, filename=attach_name + ext + "_")
 
 
+@app.get("/api/download/hashes/{hash}")
+def download_hashes(
+    hash: str, offset: int = 0, limit: int = Query(...)
+) -> Any:
```
Comment (msm-code) [inline src/app.py:68]: (i know this is WIP, but this should have a proper type in the final version)

File: src/mqueryfront/src/QueryResultsStatus.js [hunk 3]
```diff
@@ -224,10 +224,28 @@ class QueryResultsStatus extends Component {
...
+                                            downloadUrl={
+                                                API_URL +
+                                                "/download/hashes/" +
+                                                this.props.qhash +
+                                                "?limit=" +
+                                                lenMatches
+                                            }
+                                            downloadName={
+                                                "sha256_hashes" +
+                                                this.props.qhash
```
Comment (msm-code) [inline src/mqueryfront/src/QueryResultsStatus.js:246]: add `.txt`

File: src/app.py [hunk 4]
```diff
@@ -62,6 +62,21 @@ def download(job_id: str, ordinal: str, file_path: str) -> Any:
     return FileResponse(file_path, filename=attach_name + ext + "_")
 
 
+@app.get("/api/download/hashes/{hash}")
+def download_hashes(
+    hash: str, offset: int = 0, limit: int = Query(...)
```
Comment (msm-code) [inline src/app.py:67]: offset and limit parameters are probably unnecessary in this API

File: src/mqueryfront/src/components/ActionDownload.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faDownload } from "@fortawesome/free-solid-svg-icons";
+
+const ActionDownload = (props) => {
+    return (
+        <a href={props.downloadUrl} download={props.downloadName + ".txt"}>
```
Comment (msm-code) [inline src/mqueryfront/src/components/ActionDownload.js:7]: and no reason for .txt here (i.e. txt should be part of downloadname)

**Comment by msm-code** [comment]
I've tried to merge this, and ended up with this beautiful speciemen:

ss(1)

So far I've tried othis:

ss

And this:

ss

And a version with some breathing space:

ss
 
I feel like they all insult the new table design with their ugliness and feel a bit out of place, but it's the top of my artistic skill

**Pull request merged**